docs/implementation.md

```diff
@@ -61,9 +63,157 @@ records such that
+   2. Inserts Only to Read-Optimized Files - For indices like BloomFilters which are embedded with read-optimized
+      file, Hudi treats writing new records in the same way as inserting to Copy-On-Write files.
+
+As in the case of Copy-On-Write, the input tagged records are partitioned such that all upserts destined to 
+a `file id` are grouped together. This upsert-batch is written as one or more log-blocks written to log-files.
+Hudi allows clients to control log file sizes (See [Storage Configs](../configurations))
+
+The WriteClient API is same for both Copy-On-Write and Merge-On-Read writers.
+ 
+With Merge-On-Read, several rounds of data-writes would have resulted in accumulation of one or more log-files.
+All these log-files along with base-parquet (if exists) constitute a `file slice` which represents one complete version
+of the file. 
+  
+#### Compactor
+
+Realtime Readers will perform on-the-fly merge of these delta log-files to provide the most recent (committed) view of 
+the dataset. To keep the query-performance in check, Hudi supports compacting these log-files asynchronously 
+(w.r.t writer)to create read-optimized views. Asynchronous Compaction involves 2 steps:
+
+  * `Compaction Schedule` : Hudi Write Client exposes API to create Compaction plans which contains the list of `file slice`
+    to be compacted atomically in a single compaction commit. Hudi allows pluggable strategies for choosing
+    file slices for each compaction runs. This step is typically done inline by Writer process as Hudi expects
+    only one schedule is being generated at a time which allows Hudi to enforce the constraint that pending compaction
+    plans do not step on each other file-slices. This constraint allows for multiple concurrent `Compactors` to run at 
+    the same time. Some of the common strategies used for choosing `file slice` for compaction are:
+    * BoundedIO - Limit the number of file slices chosen for a compaction plan by expected total IO (read + write) 
+    needed to complete compaction run 
+    * Log File Size - Prefer file-slices with larger amounts of delta log data to be merged
+    * Day Based - Prefer file slice belonging to latest day partitions
+    ```
+        API for scheduling compaction
+          /**
+           * Schedules a new compaction instant
+           * @param extraMetadata
+           * @return Compaction Instant timestamp if a new compaction plan is scheduled
+           */
+           Optional<String> scheduleCompaction(Optional<Map<String, String>> extraMetadata) throws IOException;
+     ```
+  * `Compactor` : Hudi provides a separate API in Write Client to execute a compaction plan. The compaction
+    plan (just like a commit) is identified by a timestamp. Most of the design and implementation complexity for Async 
+    Compaction has gone for guaranteeing snapshot isolation to readers and writer when
```

Comment:
Change to : "Most of the design and implementation complexity for Async Compaction guarantees snapshot isolation to readers and writer.."